### PR TITLE
[Tune] PTL Fractional GPUs

### LINKED ISSUE #14624: [tune] What are workarounds required to use fractional GPU with Pytorch Lightning
In the tutorial, https://docs.ray.io/en/latest/tune/tutorials/tune-pytorch-lightning.html, it is mentioned that certain workarounds needed to use fraction GPU with Pytorch lightning. Would you please give me some pointers what must be done?

### PR DESCRIPTION
**Pull request by amogkam**
Closes https://github.com/ray-project/ray/issues/14624

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: doc/source/tune/_tutorials/tune-pytorch-lightning.rst [hunk 1]
```diff
@@ -228,11 +228,16 @@ we would like to use:
...
    :dedent: 4
 
-Please note that in the current state of PyTorch Lightning, training
-on :doc:`fractional GPUs </using-ray-with-gpus>` or
-multiple GPUs requires some workarounds. We will address these in a
-separate tutorial - for now this example works with no or exactly one
-GPU.
+You can also specify :doc:`fractional GPUs for Tune </using-ray-with-gpus>`, allowing multiple trials to share GPUs
+and thus increase concurrency under resource constraints. While the ``gpus_per_trial`` passed into
+Tune is a decimal value, the ``num_gpus`` passed into the ``pl.Trainer`` should still be an integer.
```
Comment (richardliaw) [inline doc/source/tune/_tutorials/tune-pytorch-lightning.rst:233]:
```suggestion
Tune is a decimal value, the ``gpus`` passed into the ``pl.Trainer`` should still be an integer.
```

File: python/ray/tune/examples/mnist_pytorch_lightning.py [hunk 2]
```diff
@@ -259,7 +262,8 @@ def tune_mnist_pbt(num_samples=10, num_epochs=10, gpus_per_trial=0):
             train_mnist_tune_checkpoint,
             data_dir=data_dir,
             num_epochs=num_epochs,
-            num_gpus=gpus_per_trial),
+            # Handle fractional GPU case. Training func should still take int.
+            num_gpus=math.ceil(gpus_per_trial)),
```
Comment (richardliaw) [inline python/ray/tune/examples/mnist_pytorch_lightning.py:266]: can we put this in `Trainer(gpus=math.ceil(...))` instead of here?

**Review by richardliaw — APPROVED** [review]
lgtm ping when tests pass

File: python/ray/tune/examples/mnist_pytorch_lightning.py (hunk 2 above)
Comment (amogkam) [reply python/ray/tune/examples/mnist_pytorch_lightning.py]: Good point, updated.

**Comment by amogkam** [comment]
@richardliaw can you merge this please

**Comment by richardliaw** [comment]
nice! thanks for the ping and the fast turnaround here.

**Pull request merged**